commit b1e6e600f3d8778e4cc70bc765560959b91ea540
Author: Fabio M. Di Nitto <fdinitto@redhat.com>
Date:   Mon Jul 26 15:17:49 2010 +0200

    cman-preconfig: better handle of logging reload operation
    
    Resolves: rhbz#615202
    
    Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

diff --git a/cman/daemon/cman-preconfig.c b/cman/daemon/cman-preconfig.c
index fe12004..0de1f04 100644
--- a/cman/daemon/cman-preconfig.c
+++ b/cman/daemon/cman-preconfig.c
@@ -1213,7 +1213,7 @@ static int cmanpre_reloadconfig(struct objdb_iface_ver0 *objdb, int flush, const
 	objdb->object_find_create(OBJECT_PARENT_HANDLE, "logging", strlen("logging"), &find_handle);
 	ret = objdb->object_find_next(find_handle, &object_handle);
 	objdb->object_find_destroy(find_handle);
-	if (ret) {
+	if (!ret) {
 		objdb->object_destroy(object_handle);
 	}
 
@@ -1226,6 +1226,8 @@ static int cmanpre_reloadconfig(struct objdb_iface_ver0 *objdb, int flush, const
 	/* copy /cluster/totem to /totem */
 	ret = copy_tree_to_root(objdb, "totem", 0);
 
+	add_cman_overrides(objdb);
+
 	return 0;
 
 err:
